app/src/test/java/com/blackboxteam/app/blackbox/UserSessionPersistenceTest.java

```diff
@@ -0,0 +1,54 @@
+package com.blackboxteam.app.blackbox;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.blackboxteam.app.blackbox.util.UserSessionPersistence;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for UserSessionPersistenceClass
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class UserSessionPersistenceTest {
+
+    private UserSessionPersistence mMockUserSessionPersistence;
+
+    @Mock
+    Context mMockContext;
+
+    UserSessionPersistence userSessionPersistence = new UserSessionPersistence(mMockContext);
+
+
+    @Test
+    public void saveUserInfo_shouldSaveUserInfo(){
+        SharedPreferences mockedSharedPreference = Mockito.mock(SharedPreferences.class);
+
+        Mockito.when(mMockContext.getSharedPreferences("userInfo", Context.MODE_PRIVATE)).thenReturn(mockedSharedPreference);
+
+        SharedPreferences.Editor mockedEditor = Mockito.mock(SharedPreferences.Editor.class);
```

Comment:
Move all intialization to the top, before defining their behaviours.

Reply:
Done